### PR TITLE
instr(statsd): Low-cardinality metric name tag

### LINKED ISSUE #1837: Stop tagging statsd metrics by metric name
We currently set the metric name on at least two statsd metrics:

https://github.com/getsentry/relay/blob/8f490056800d879903c669f7838f00b97fd72bf2/relay-metrics/src/aggregation.rs#L1706

This tag is no longer low-cardinality because of custom measurements.

Possible solution: Make metric name an `enum` with an `Other(String)` variant. Map that variant to `"other"` for the statsd tag.

### PR DESCRIPTION
**Pull request by jjbayer**
Instead of allowing _any_ metric name as a statsd tag, limit it to a predefined set of well known names.

Fixes #1837.

#skip-changelog

### CODE REVIEW

File: relay-metrics/src/statsd.rs [hunk 1]
```diff
@@ -148,3 +148,33 @@ impl GaugeMetric for MetricGauges {
...
+        "s:sessions/error@none",
+        "d:transactions/duration@millisecond",
+        "s:transactions/user@none",
+        "c:transactions/count_per_root_project@none",
+    ]
+    .into_iter()
+    .find(|s| s == &metric_name)
+    {
+        return known_name;
+    }
```
Comment (olksdr) [inline relay-metrics/src/statsd.rs:170]: Why not just use `contains` method on slice here?
```suggestion
    let metric_name = value.as_ref();
    if [
        "c:sessions/session@none",
        "s:sessions/user@none",
        "s:sessions/error@none",
        "d:transactions/duration@millisecond",
        "s:transactions/user@none",
        "c:transactions/count_per_root_project@none",
    ].contains(&metric_name)
    {
        return metric_name;
    }
```

**Review by olksdr — APPROVED** [review]
I have one questions , otherwise lgtm!

File: relay-metrics/src/statsd.rs (hunk 1 above)
Comment (olksdr) [reply relay-metrics/src/statsd.rs:170]: you might have to look into lifetimes, but it should also work just fine.

Comment (jjbayer) [reply relay-metrics/src/statsd.rs:170]: I first used `.contains()`, but `metric_name` does not have `'static` lifetime, which is why I return a reference to the string literal instead.

**Pull request merged**